web/html/src/components/input/InputBase.tsx

```diff
@@ -173,7 +177,10 @@ export class InputBase<ValueType = string> extends React.Component<InputBaseProp
     }
 
     if (!this.props.disabled && (value || this.props.required)) {
-      if (this.props.required && !value) {
+      const noValue = Array.isArray(this.props.name) ?
+        value == null || Object.values(value).filter(v => v).length === 0 :
```

Comment:
`filter(v => v)` is a dangerous way to solve this since `InputBase` is a generic. We currently don't have such a case, but it would be completely valid to use `InputBase` in a configuration where the value type is a number or boolean, in which this filter will result in a bug.  

As a fix I would recommend adding a method to the class a-la `isEmptyValue` that's something along these lines:  
```ts
isEmptyValue(input: any) {
  if (typeof input === 'string') {
    return input.trim() === '';
  }
  return isNil(input);
}
```

Reply:
OK